warehouse/query-core/src/main/java/datawave/query/jexl/functions/EvaluationPhaseFilterFunctions.java

```diff
@@ -145,6 +146,29 @@ public static FunctionalSet<ValueTuple> isNotNull(Object fieldValue) {
                 return FunctionalSet.singleton(getHitTerm(fieldValue));
             }
         }
+        
+        return FunctionalSet.emptySet();
+    }
+    
+    /**
+     * Returns a {@link FunctionalSet} of hit terms found for {@code fieldValue1}. If {@code fieldValue1} is a singular value tuple, a singleton
+     * {@link FunctionalSet} with the hit term from it will be returned. If {@code fieldValue1} and {@code fieldValue2} are non-empty collections of value
+     * tuples, a {@link FunctionalSet} containing the hit terms from each value in the {@code fieldValue1} collection will be returned. Otherwise, an empty
+     * {@link FunctionalSet} will be returned.
+     *
+     * @param fieldValue1
+     *            1st field value to evaluate.
+     * @param fieldValue2
+     *            2nd field value to evaluate.
+     * @return the {@link FunctionalSet} of hit terms found
+     */
+    public static FunctionalSet<ValueTuple> isNotNull(Object fieldValue1, Object fieldValue2) {
```

Comment:
Hey Drew, thank you so much for the extra guidance!  This ticket makes a lot more sense now.  

I have one quick question...I'm struggling with the return type for the isNotNull methods.  Does the return type need to be of type, "FunctionalSet<ValueTuple>"?  To me, it seems that a True/False would be a better response from the isNotNull methods, but I also do not have a great understanding of how query side of the house either...  Any direction on this return type would be much appreciated.  The isNull is just a boolean, so that one makes good sense to me.

Thanks Buddy!

Reply:
I think for the sake of consistency, the iterable version of isNotNull method will have to return a functional set just like it's single-fielded counterpart.

Reply:
Would a List<FunctionalSet> for the return type work?  That way I could put the FuctionalSet returned by the original method in a List for each item in the Iterable parameter passed in.  Let me know when you can, thanks!

Reply:
Modified code to utilize more appropriate method signature with multi-fielded filter parameters for the isNull and isNotNull overloaded methods.  Updated method signatures now utilize "Iterable<?> iterable" parameters.  Updated unit tests to test new method signatures and logic.

Reply:
Please review as time allows.  Thanks!